base/linalg/lapack.jl

```diff
@@ -1256,6 +1512,57 @@ for (geev, gesvd, gesdd, ggsvd, elty, relty) in
         end
     end
 end
+
+"""
+    geev!(jobvl, jobvr, A) -> (W, VL, VR)
+
+Finds the eigensystem of `A`. If `jobvl = N`, the left eigenvectors of
+`A` aren't computed. If `jobvr = N`, the right eigenvectors of `A`
+aren't computed. If `jobvl = V` or `jobvr = V`, the corresponding
+eigenvectors are computed. Returns the eigenvalues in `W`, the right
+eigenvectors in `VR`, and the left eigenvectors in `VL`.
+"""
+geev!(jobvl::Char, jobvr::Char, A::StridedMatrix)
+
+"""
+    gesdd!(job, A) -> (U, S, VT)
+
+Finds the singular value decomposition of `A` using a divide and conquer
+approach. If `job = A`, all the columns of `U` and the rows of `V'` are
+computed. If `job = N`, no columns of `U` or rows of `V'` are computed.
+If `job = O`, `A` is overwritten with the columns of (thin) `U` and the
+rows of (thin) `V'`. If `job = S`, the columns of (thin) `U` and the rows
+of (thin) `V'` are computed and returned separately.
+"""
+gesdd!(job::Char, A::StridedMatrix)
+
+"""
+    gesvd!(jobu, jobvt, A) -> (U, S, VT)
+
+Finds the singular value decomposition of `A`. If `jobu = A`, all the
+columns of `U` are computed. If `jobvt = A` all the rows of `V'` are
+computed. If `jobu = N`, no columns of `U` are computed. If `jobvt = N`
+no rows of `V'` are computed. If `jobu = O`, `A` is overwritten with the
+columns of (thin) `U`. If `jobvt = O`, `A` is overwritten with the rows
+of (thin) `V'`. If `jobu = S`, the columns of (thin) `U` are computed
+and returned separately. If `jobvt = S` the rows of (thin) `V'` are
+computed and returned separately. `jobu` and `jobvt` can't both be `O`.
+
+Returns `U`, `S`, and `Vt`, where `S` are the singular values of `A`.
+"""
+gesvd!(jobu::Char, jobvt::Char, A::StridedMatrix)
+
+"""
+    ggsvd!(jobu, jobv, jobq, A, B) -> (U, V, Q, alpha, beta, k, l, R)
+
+Finds the generalized singular value decomposition of `A` and `B`.
```

Comment:
would be cool to spell out the matrix equation that this (and the normal svd's above) are solving